### PR TITLE
[NS-534]  Use boto sessions for S3 operations. Smart open update to 1.8.3.

### PR DESCRIPTION
**Pull request by sandeepmjay**
Update smart open to 1.8.3
Use sts to create boto sessions for S3 authentication
Update tests to use mock_sts
update config.py to include APP_ROLE
Parameterize Encryption on S3 to use AES256

https://jira.ets.berkeley.edu/jira/browse/NS-534

### CODE REVIEW

File: nessie/externals/s3.py [hunk 1]
```diff
@@ -204,14 +216,20 @@ def upload_from_url(url, s3_key, on_stream_opened=None):
...
                     'ContentEncoding': 'gzip',
                     'ContentType': 'text/plain',
                 })
+            session = get_session()
             # smart_open needs to be told to ignore the .gz extension, or it will smartly attempt to double-compress it.
-            with smart_open.smart_open(s3_url, 'wb', ignore_extension=True, s3_upload_args=s3_upload_args) as s3_out:
+            with smart_open.open(
+                s3_url,
+                'wb',
+                ignore_ext=True,
```
Comment (johncrossman) [inline nessie/externals/s3.py]: @sandeepmjay `ignore_ext` is right and `ignore_extension` is wrong?

**Comment by johncrossman** [comment]
Looks good @sandeepmjay  I put one question.

File: nessie/externals/s3.py (hunk 1 above)
Comment (sandeepmjay) [reply nessie/externals/s3.py]: Smart open folks thought it would optimize to maximize the smartness with the change in 1.8.3

Comment (johncrossman) [reply nessie/externals/s3.py]: Smart they are.

**Pull request merged**